src/Core/Managed/Shared/TelemetryClient.cs

```diff
@@ -325,56 +325,65 @@ public void Track(ITelemetry telemetry)
             // It is hidden from intellisense to prevent customer confusion.
             if (this.IsEnabled())
             {
-                string instrumentationKey = this.Context.InstrumentationKey;
+                this.Initialize(telemetry);
 
-                if (string.IsNullOrEmpty(instrumentationKey))
-                {
-                    instrumentationKey = this.configuration.InstrumentationKey;
-                }
+                telemetry.Sanitize();
+
+                this.Channel.Send(telemetry);
 
-                if (string.IsNullOrEmpty(instrumentationKey))
+                if (System.Diagnostics.Debugger.IsAttached)
                 {
-                    return;
+                    this.WriteTelemetryToDebugOutput(telemetry);
                 }
+            }
+        }
 
-                var telemetryWithProperties = telemetry as ISupportProperties;
-                if (telemetryWithProperties != null)
-                {
-                    if (this.Channel.DeveloperMode.HasValue && this.Channel.DeveloperMode.Value)
-                    {
-                        if (!telemetryWithProperties.Properties.ContainsKey("DeveloperMode"))
-                        {
-                            telemetryWithProperties.Properties.Add("DeveloperMode", "true");
-                        }
-                    }
+        /// <summary>
+        /// This method is an internal part of Application Insights infrastructure. Do not call.
+        /// </summary>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public void Initialize(ITelemetry telemetry)
+        {
+            string instrumentationKey = this.Context.InstrumentationKey;
 
-                    Utils.CopyDictionary(this.Context.Properties, telemetryWithProperties.Properties);
-                }
+            if (string.IsNullOrEmpty(instrumentationKey))
+            {
+                instrumentationKey = this.configuration.InstrumentationKey;
+            }
+
+            if (string.IsNullOrEmpty(instrumentationKey))
+            {
+                return;
```

Comment:
```TelemetryClient``` has a private ``TelemetryConfiguration```. We do not want to expose it as it will create confusion - many clients shares the same telemetry configuration object and changing one you wouldn't expect that another will be changed as well.

Now, when you want to build extension for TelemetryClient - we found out that we need to have access to telemetry initializers related to this client. So instead of exposing configuration for telemetry client we propose to expose this ```Initialize``` method.

For example , we are building Operation API. We want to build it as an extension for TelemetryClient. For operation API we need to initializer operation on the thread where operation was started. So we need to call initializers. If we will put operations inside the Core package - it will not be a problem. But before we'll do it - this is a nice solution for the problem.